features/bevragen/raadpleeg-met-burgerservicenummer/fout-cases.feature

```diff
@@ -0,0 +1,131 @@
+
+Functionaliteit: Raadpleeg met burgerservicenummer - fout cases
+
+Rule: De burgerservicenummer parameter is een verplichte parameter
+
+  @fout-case
+  Scenario: De burgerservicenummer parameter is niet opgegeven
+    Als personen wordt gezocht met de volgende parameters
+    | naam   | waarde                          |
+    | type   | RaadpleegMetBurgerservicenummer |
+    | fields | burgerservicenummer             |
+    Dan heeft de response een object met de volgende gegevens
+    | naam     | waarde                                                      |
+    | type     | https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.1 |
+    | title    | Minimale combinatie van parameters moet worden opgegeven.   |
+    | status   | 400                                                         |
+    | detail   | De foutieve parameter(s) zijn: burgerservicenummer.         |
+    | code     | paramsCombination                                           |
+    | instance | /haalcentraal/api/brp/personen                              |
+    En heeft het object de volgende 'invalidParams' gegevens
+    | code     | name                | reason                  |
+    | required | burgerservicenummer | Parameter is verplicht. |
+
+Rule: De burgerservicenummer parameter bevat een lijst met minimaal één burgerservicenummer
+
+  @fout-case
+  Abstract Scenario: De burgerservicenummer parameter bevat een lege lijst
+    Als personen wordt gezocht met de volgende parameters
+    | naam                | waarde                          |
+    | type                | RaadpleegMetBurgerservicenummer |
+    | burgerservicenummer |                                 |
+    | fields              | burgerservicenummer             |
+    Dan heeft de response een object met de volgende gegevens
+    | naam     | waarde                                                      |
+    | type     | https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.1 |
+    | title    | Een of meerdere parameters zijn niet correct.               |
+    | status   | 400                                                         |
+    | detail   | De foutieve parameter(s) zijn: burgerservicenummer.         |
+    | code     | paramsValidation                                            |
+    | instance | /haalcentraal/api/brp/personen                              |
+    En heeft het object de volgende 'invalidParams' gegevens
+    | code     | name                | reason                          |
+    | minItems | burgerservicenummer | Array bevat minder dan 1 items. |
+
+  @fout-case
+  Scenario: De burgerservicenummer parameter bevat een string van burgerservicenummers gescheiden door een komma
+    Als personen wordt gezocht met de volgende parameters
+    | naam                             | waarde                          |
+    | type                             | RaadpleegMetBurgerservicenummer |
+    | burgerservicenummer (als string) | 999994086,999994086,999994086   |
```

Comment:
Met de formulering '(als string)' kun je dus de manier beïnvloeden waarop de test automation code de waarde er achter moet interpreteren?

Reply:
ja, in de automation code wordt hierop gecheckt. Bij deze naam wordt de waarde niet omgezet naar een array